src/sqlfluff/core/rules/base.py

```diff
@@ -258,51 +258,55 @@ def create_after(
         """Create edit segments after the supplied anchor segment."""
         return cls("create_after", anchor_segment, edit_segments, source)
 
-    def has_template_conflicts(self, templated_file: TemplatedFile) -> bool:
```

Comment:
The diffs below may be easier to read if you keep in mind that I split `has_template_conflicts()` into two functions:
* `get_fix_slices()`
* `has_template_conflicts()`